machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +54,22 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
+	// GuestAccelerators is a list of GPUs to be attached to the VM.
+	// +optional
+	GuestAccelerators []*GCPAcceleratorConfig `json:"guestAccelerators,omitempty"`
 	// Preemptible indicates if created instance is preemptible
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance determines the behavior when a maintenance event occurs that might cause the instance to reboot.
+	// [default] MIGRATE - causes Compute Engine to live migrate an instance when there is a maintenance event.
+	// TERMINATE - stops an instance instead of migrating it.
+	// This is required to be set to TERMINATE if you want to provision machine with attached GPUs.
+	// +optional
+	OnHostMaintenance string `json:"onHostMaintenance,omitempty"`
+	// AutomaticRestart determines behavior when an instance crashes or is stopped due to maintenance event (default true).
```

Comment:
```suggestion
	// AutomaticRestart determines the behavior when an instance crashes or is stopped due to a maintenance event (default true).
```
Thanks, this is a little clearer.  Is it going to be obvious to people using this API what "a maintenance event" means though, or who or what could initiate a "maintenance event"?  Would it be accurate to say "when an instance crashes or the underlying infrastructure provider stops the instance as part of a maintenance event"?

Reply:
I thought it is clear enough. But sure it cant hurt to specify it even better.